src/Momento.Sdk/Requests/Vector/MetadataFields.cs

```diff
@@ -0,0 +1,56 @@
+using System.Collections.Generic;
+
+namespace Momento.Sdk.Requests.Vector;
+
+/// <summary>
+/// Wrapper for a list of metadata fields. Used in vector methods that can either take a
+/// list of metadata to look up, or a value specifying that all metadata should be returned.
+/// </summary>
+public abstract class MetadataFields
+{
+    /// <summary>
+    /// Static value representing all metadata fields.
+    /// </summary>
+    public static readonly AllFields All = new AllFields();
+
+    /// <summary>
+    /// Implicitly convert a list of strings to a MetadataFields. Allows for passing a bare list instead
+    /// of having to explicitly create a MetadataFields object.
+    /// </summary>
+    /// <param name="fields">The fields to look up.</param>
+    /// <returns></returns>
+    public static implicit operator MetadataFields(List<string> fields) => new List(fields);
```

Comment:
neat, I didn't know you could do that.

You can't put an enumerable as the argument though?

Reply:
Nope, it has to be a specific implementation. The best we can do is add convers for all implementations that make sense.

Reply:
i see.

i actually think list is probably fine, especially for MVP. was just curious because I hadn't seen this feature before.